lib/Sema/CSDiagnostics.cpp

```diff
@@ -3399,6 +3402,38 @@ bool MissingMemberFailure::diagnoseForDynamicCallable() const {
   return false;
 }
 
+bool MissingMemberFailure::diagnoseForDefaultAnyInCollection() const {
+  auto &cs = getConstraintSystem();
+  auto *expr = getAsExpr(getAnchor());
```

Comment:
Nit: This should be a `castToExpr`.

Reply:
When is the case to use each `castToExpr` and `getAsExpr`?

Reply:
Same as `dyn_cast` vs. `cast`, if you don't know whether anchor is an expression it's better to use `getAsExpr` to check and even cast it to particular kind e.g. you can do `getAsExpr<UnresolvedMemberExpr>(getAnchor())`  here and `if (!expr) return false;`.

Reply:
Nice, Thanks 👍